manage

```diff
@@ -536,9 +537,24 @@ ${DIRECTORY}/icons/none-1.png
   (sleep 5; kill $yadpid 2>/dev/null) &
   
   #diagnose every failed app's logfile - list item format is $action;$app;$exitcode
-  failed_apps="$(echo "$queue" | grep ';1$' | awk -F';' '{print $2}')"
+  failed_apps="$(echo "$queue" | grep ';[0-9]\+$' | grep -v ';0$' | awk -F';' '{print $2}')"
   diagnose_apps "$failed_apps"
 
+  successful_apps="$(echo "$queue" | grep ';0$' | awk -F';' '{print $2}')"
+  if [ ! -z "${successful_apps}" ]; then
+    yad "${yadflags[@]}" --class Pi-Apps --name "Pi-Apps" --width=700 --height=300 --title="Pi-Apps install/uninstall/updates complete" \
+      --image="${DIRECTORY}/icons/logo-64.png" --image-on-top \
+      --text='Thanks for using Pi-Apps!'$'\n'"The following apps were succcessful: $successful_apps" \
+      --wrap --fontname=12 \
+      --button='Ok'
```

Comment:
This is the dialog I was suggesting that we have present the "thank for using pi-apps" text with a call to action and the icons of the apps that were successfully installed/uninstalled/updated.

I think we had some code that presented all the app icons on one screen in the history of pi-apps somewhere. It would be interesting to do that but also show a small icon in the top left of each app icon for installing/uninstalling/updating.

Reply:
> I think we had some code that presented all the app icons on one screen in the history of pi-apps somewhere.

Yeah this:

https://github.com/Botspot/pi-apps/commit/d169e2aabb3ca14ede7ccad08c7eb0ac846d19f3
https://github.com/Botspot/pi-apps/commit/7191be328f531c888cae9738f1ff4c530719cd79
https://github.com/Botspot/pi-apps/commit/b7975c0ca6721345a576d5af009c552df4bf7b8e

And I removed it as part of this because we were using the splashscreen to hide the fact that (at the time) pi-apps was slow to start:

https://github.com/Botspot/pi-apps/commit/7cd4f68d0d3afad90b6d34d2d25d0b4a45947cf0

So we can bring that splashscreen code back (with modifications) for use in finish dialog.